sdk/textanalytics/Azure.AI.TextAnalytics/src/OperationIdInformation.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Azure.AI.TextAnalytics
+{
+    internal class OperationIdInformation
+    {
+        public OperationIdInformation() { }
+
+        public OperationIdInformation(string jobId, IDictionary<string, int> indexMap, bool? showStats = default)
+        {
+            JobId = jobId;
+            InputDocumentOrder = new Dictionary<string, int>(indexMap);
+            ShowStats = showStats;
+        }
+
+        public Dictionary<string, int> InputDocumentOrder { get; set; }
+        public bool? ShowStats { get; set; }
+        public string JobId { get; set; }
+    }
```

Comment:
Are customers serializing these operation IDs across services (i.e., throw it in a Storage Queue so somebody else can wait for it to finish)?  If so, should we also add a synthetic version so we can tell whether we're correctly representing one of them?  Should we also add an AdditionalInfo dictionary with https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonextensiondataattribute?view=net-5.0 on it to make sure nothing else is present in the string?

Reply:
And are we doing the same thing across all languages?  Could I take one of these synthetic IDs from .NET and use it with a Java or Python client library?

Reply:
^ that's correct for Python. We pickle/serialize the entire initial response (POST) object, though, not just the ID of the operation. So no, a continuation token from Python would not work for other languages, or vice versa. I'm not sure if the idea of sharing these Ids across languages came up at design time...

Reply:
> Are customers serializing these operation IDs across services (i.e., throw it in a Storage Queue so somebody else can wait for it to finish)? If so, should we also add a synthetic version so we can tell whether we're correctly representing one of them?

Not sure I follow. The operationID only makes sense to rehydrate an LRO. Unless a user wants to use REST they shouldn't be looking inside the operationId value itself, right?

> If so, should we also add a synthetic version so we can tell whether we're correctly representing one of them?

What do you mean with a `synthetic version`?

Reply:
A somewhat common pattern for cloud native microservices is to have one worker kick off the LRO, stash the operation ID in something like a Storage Queue, and have other workers pull operation IDs out of the queue to check if they're finished to resume processing asynchronously.  That separation between worker 1 and worker 2 can cause lots of fun problems.  Maybe worker 1 was a frontend written in JS and worker 2 is an Azure Function in C#?  Or maybe worker 1 and worker 2 are using different features of the service and are upgraded to different versions of the client library?  How can we design our operation IDs in a way that enables those customer scenarios?

When we did [something similar for Search continuation tokens](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/src/Models/SearchContinuationToken.cs), we also threw in the api-version as a "synthetic version" to distinguish "this is how continuation tokens looked in XYZ version of the service" in case they start looking different in a future version of the service.  If someone gives us a version we don't know about, we can fail with a clear error.  We also aligned on that pattern across languages in case you started searching in one language and wanted to resume in another - an admittedly unlikely scenario, but one that was easy enough to support once we were committed to making continuation tokens version tolerant.

Reply:
I have added the api-version to the continuation token body so we can throw whenever it changes. I agree that throwing a meaningful error message is better than silently doing something the user may not be expecting.

> ^ that's correct for Python. We pickle/serialize the entire initial response (POST) object, though, not just the ID of the operation. So no, a continuation token from Python would not work for other languages, or vice versa. I'm not sure if the idea of sharing these Ids across languages came up at design time...

As Krista pointed out, cross-language compatibility may be a bit hard to achieve since Python LROs serialize the whole initial response when serializing their continuation token. I still need to get confirmation from Java and JS and see if we can do something for other languages, though.